identities/models.py

```diff
@@ -44,3 +48,74 @@ class Identity(models.Model):
 
     def __str__(self):
         return str(self.id)
+
+    def remove_details(self, user):
+        updated_details = {}
+        for attribute, value in self.details.iteritems():
+            updated_details[attribute] = "removed"
+        self.details = updated_details
+        self.updated_at = datetime.datetime.now()
+        self.updated_by = user
+        self.save()
+
+
+class OptOut(models.Model):
+    """An optout"""
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False,
+                          help_text="UUID this optout request.")
+    identity = models.ForeignKey(Identity, null=True,
+                                 help_text="UUID for the identity opting out.")
```

Comment:
Unfortunately, setting `null=False` causes the serializer to force the user to provide a value for it. So if we want them to be able to optout by giving an address then it needs to be able to be null. But i've accounted for it where I now get the id